From c47b607ed68a9ef0ef6584e6b7bb41599da31081 Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Thu, 13 Jul 2023 12:22:30 +0530
Subject: [PATCH] wifi: ath12k: fix incorrect phymode in
 ath12k_wmi_put_wmi_channel

ath12k_wmi_put_wmi_channel contains check for MODE_11AX_HE160 and MODE_11AC_VHT80_80
which is not a valid check as phymode is always set as EHT in
ath12k_mac_monitor_vdev_start, ath12k_mac_vdev_start_restart and
ath12k_mac_multi_vdev_restart

ath12k driver always comes up in EHT phy mode.
So change the check in ath12k_wmi_put_wmi_channel to EHT phy mode.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/wmi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/wmi.c b/drivers/net/wireless/ath/ath12k/wmi.c
index e9a3ae1e21bd..471222dfbda3 100644
--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -1155,7 +1155,7 @@ static void ath12k_wmi_put_wmi_channel(struct wmi_channel *chan,
 
 		chan->band_center_freq2 = chan_arg.band_center_freq1;
 
-	} else if ((chan_arg.mode == MODE_11AX_HE160) || (chan_arg.mode == MODE_11BE_EHT160)) {
+	} else if (chan_arg.mode == MODE_11BE_EHT160) {
 		if (chan_arg.freq > chan_arg.band_center_freq1)
 			chan->band_center_freq1 = center_freq1 + 40;
 		else
@@ -1163,7 +1163,7 @@ static void ath12k_wmi_put_wmi_channel(struct wmi_channel *chan,
 
 		chan->band_center_freq2 = chan_arg.band_center_freq1;
 
-	} else if (chan_arg.mode == MODE_11AC_VHT80_80) {
+	} else if (chan_arg.mode == MODE_11BE_EHT80_80) {
 		chan->band_center_freq2 = chan_arg.band_center_freq2;
 	} else {
 		chan->band_center_freq2 = 0;
-- 
2.7.4

